datafusion/functions-aggregate/src/approx_median.rs

```diff
@@ -116,4 +112,8 @@ impl AggregateUDFImpl for ApproxMedian {
             acc_args.exprs[0].data_type(acc_args.schema)?,
         )))
     }
+
+    fn coerce_types(&self, _arg_types: &[DataType]) -> Result<Vec<DataType>> {
+        Ok(vec![DataType::Float64])
```

Comment:
OOC, why this instead of just defining the signature as DataType::Float64? Afaik DF already tries to coerce inputs to the signature

Reply:
I think you are referring to `coerced_from`? I want to deprecate that function because the downside of having single truth of coercion make updating the coercion rule unpredictable, easy to cause bug without notice (hard to have full test coverage too). The new approach is to handle the coercion by signature.